### ISSUE TITLE
Coriant login banner requires confirmation

### ISSUE THREAD

**Issue by ricardobendita** [ktbyers/netmiko#1730]
I have the script:

infinera = {
	"device_type": "coriant",
	"ip": "10.251.64.39",
	"username": "test",
	"port": 22,
	"password": "test",
}

net_connect = netmiko.ConnectHandler(**infinera)
net_connect.send_command("enable")
net_connect.send_command("terminal length 200")
output = net_connect.send_command("show ip int bri")
print(output)

But I have the next error message abour prompt:
SSH connection established to 10.251.64.39:22
Interactive SSH session established
Traceback (most recent call last):
  File "netmiko_huawei.py", line 24, in <module>
    net_connect = netmiko.ConnectHandler(**infinera)
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\ssh_dispatcher.py", line 259, in ConnectHandler
    return ConnectionClass(*args, **kwargs)
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 327, in __init__
    self._open()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 333, in _open
    self._try_session_preparation()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 756, in _try_session_preparation
    self.session_preparation()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\coriant\coriant_ssh.py", line 7, in session_preparation
    self.set_base_prompt()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\coriant\coriant_ssh.py", line 37, in set_base_prompt
    delay_factor=delay_factor,
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 1079, in set_base_prompt
    prompt = self.find_prompt(delay_factor=delay_factor)
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 1122, in find_prompt
    raise ValueError(f"Unable to find prompt: {prompt}")
ValueError: Unable to find prompt:

When I start one connection independently to device it show me:

################################################################
#      ATENCION: ACCESO PERMITIDO SOLO A PERSONAL AUTORIZADO.  #
#                DESCONECTESE INMEDIATAMENTE                   #
# ATTENTION: AUTHORIZED PERSONAL ONLY. DISCONNECT IMMEDIATELY. #
#                  AUTENTICACION VIA TACACS                    #
################################################################
~
Answer: y(es)/n(o) 

*********************************************************************
*                                                                   *
*                   Coriant 8615 Network Element                    *
*                                                                   *
*********************************************************************
Enter configuration commands, one per line. End with ^Z
hostname>

Please help me with this script.

Regards,
Ricardo.

**Comment by nullbitzero** [comment]
@ricardobendita 
you can the following:
For the answer you can include net_connect.write_channel ('yes' or 'y')
Then some conditionals or net_connect.enable ()

Luck

**Comment by ricardobendita** [comment]
@nullbitzero, thank for your answer, I try with this:

net_connect = netmiko.ConnectHandler(**infinera)
net_connect.enable()
net_connect.write_channel('y')
net_connect.send_command("enable")
net_connect.send_command("terminal length 200")
output = net_connect.send_command("show ip int bri")

But, the output is:

SSH connection established to 10.251.64.39:22
Interactive SSH session established
Traceback (most recent call last):
  File "netmiko_huawei.py", line 22, in <module>
    net_connect = netmiko.ConnectHandler(**infinera)
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\ssh_dispatcher.py", line 259, in ConnectHandler
    return ConnectionClass(*args, **kwargs)
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 327, in __init__
    self._open()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 333, in _open
    self._try_session_preparation()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 756, in _try_session_preparation
    self.session_preparation()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\coriant\coriant_ssh.py", line 7, in session_preparation
    self.set_base_prompt()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\coriant\coriant_ssh.py", line 37, in set_base_prompt
    delay_factor=delay_factor,
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 1079, in set_base_prompt
    prompt = self.find_prompt(delay_factor=delay_factor)
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 1122, in find_prompt
    raise ValueError(f"Unable to find prompt: {prompt}")
ValueError: Unable to find prompt:_

**Comment by nullbitzero** [comment]
@ricardobendita 
Modify your scrip with this:
net_connect = netmiko.ConnectHandler(**infinera)
net_connect.write_channel('y')
net_connect.send_command("enable") 
net_connect.send_command("terminal length 200")
output = net_connect.send_command("show ip int bri")

**Comment by ricardobendita** [comment]
@nullbitzero 
I try the commands that you say but the answer is the same:

_SSH connection established to 10.251.64.39:22
Interactive SSH session established
Traceback (most recent call last):
  File "netmiko_huawei.py", line 25, in <module>
    net_connect = netmiko.ConnectHandler(**infinera)
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\ssh_dispatcher.py", line 259, in ConnectHandler
    return ConnectionClass(*args, **kwargs)
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 327, in __init__
    self._open()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 333, in _open
    self._try_session_preparation()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 756, in _try_session_preparation
    self.session_preparation()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\coriant\coriant_ssh.py", line 7, in session_preparation
    self.set_base_prompt()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\coriant\coriant_ssh.py", line 37, in set_base_prompt
    delay_factor=delay_factor,
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 1079, in set_base_prompt
    prompt = self.find_prompt(delay_factor=delay_factor)
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 1121, in find_prompt
    raise ValueError(f"Unable to find prompt: {prompt}")
ValueError: Unable to find prompt:_

In my dictiornary "infinera" add the option "default_enter" of the next form:

_infinera = {
	"device_type": "coriant",
	"ip": "10.251.64.39",
	"username": "test",
	"port": 22,
	"password": "test",
	"verbose": True,
	"default_enter": "y \r\n",
}_

And the result is:

SSH connection established to 10.251.64.39:22
Interactive SSH session established
Traceback (most recent call last):
  File "netmiko_huawei.py", line 25, in <module>
    net_connect = netmiko.ConnectHandler(**infinera)
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\ssh_dispatcher.py", line 259, in ConnectHandler
    return ConnectionClass(*args, **kwargs)
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 327, in __init__
    self._open()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 333, in _open
    self._try_session_preparation()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 756, in _try_session_preparation
    self.session_preparation()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\coriant\coriant_ssh.py", line 7, in session_preparation
    self.set_base_prompt()
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\coriant\coriant_ssh.py", line 37, in set_base_prompt
    delay_factor=delay_factor,
  File "C:\Users\ASUS\AppData\Local\Programs\Python\Python37-32\lib\site-packages\netmiko\base_connection.py", line 1081, in set_base_prompt
    raise ValueError(f"Router prompt not found: {repr(prompt)}")
ValueError: Router prompt not found: 'LAL-EAL_IN1_HL5_ElAlambre>\x1b[K'

It show me the prompt "LAL-EAL_IN1_HL5_ElAlambre>" but script send me Exception ValueError, I dont understand, any idea?, thanks for any your comments

Ricardo.

**Comment by nullbitzero** [comment]
Attach the problem logs
https://github.com/ktbyers/netmiko/blob/develop/COMMON_ISSUES.md#enable-netmiko-logging-of-all-reads-and-writes-of-the-communications-channel

**Comment by ricardobendita** [comment]
Here the result :

[test.log](https://github.com/ktbyers/netmiko/files/4642799/test.log)

**Comment by ricardobendita** [comment]
Hi @nullbitzero:

I test with the next options in my dictionary:

infinera = {
	"device_type": "coriant",
	"ip": "10.251.64.39",
	"username": "test",
	"port": 22,
	"password": "test",
	"verbose": True,
}

logging.basicConfig(filename='test.log', level=logging.DEBUG)
logger = logging.getLogger("netmiko")

net_connect = netmiko.ConnectHandler(**infinera)
net_connect.write_channel('y')
net_connect.send_command("enable")
net_connect.send_command("terminal length 200")
output = net_connect.send_command("show ip int bri")
print(output)

In the result of logging I don't see that script write "y" with :

net_connect.write_channel('y')

The result logging:

[test.log](https://github.com/ktbyers/netmiko/files/4643009/test.log)

Why my script never send that "y"?.

**Comment by ktbyers** [comment]
You would have to get rid of the banner confirmation or do some special handler:

```
Answer: y(es)/n(o) [K
```

Netmiko will automatically do things after establishing the SSH connection and before you ever do the `write_channel('y')`.

You could probably try to the following:

```python
infinera = {
"device_type": "terminal_server",    # notice changed
"ip": "10.251.64.39",
"username": "test",
"port": 22,
"password": "test",
"verbose": True,
}

logging.basicConfig(filename='test.log', level=logging.DEBUG)
logger = logging.getLogger("netmiko")

net_connect = netmiko.ConnectHandler(**infinera)
net_connect.write_channel('y')
net_connect.find_prompt()
```

And see if that works. The terminal_server device_type in Netmiko could also be called `generic` it doesn't do anything after establishing the SSH connection.

If that works, you can then look at the `redispatch` function call to switch the class (search GitHub issues on this).

In other words, establish the connection using `terminal_server` device type and once you get past the banner prompting, then switch to coriant device_type.

**Comment by ricardobendita** [comment]
Hi @ktbyers, thank for your comments, I did your recommendation and I have the next result: 

[test.log](https://github.com/ktbyers/netmiko/files/4645293/test.log)

Any idea more please.

**Comment by ktbyers** [comment]
Probably and a `read_channel()` before the write_channel potentially with a loop and a time.sleep.

Then add a conditional and make sure you are seeing the `Answer: y(es)/n(o)` message before sending the `y`.

You also might need to verify whether it requires an `enter` at the end of it.

i.e. should it be `write_channel("y\n")`
